components/chat/chatBar.tsx

```diff
@@ -42,7 +42,16 @@ const ChatBar = ({
     messages,
     setMessages,
   } = useContext(ChatContext);
-  const [userMessages, setUserMessages] = useState<string[]>([]);
+
+  const messagesRef = useRef(messages);
+  messagesRef.current = messages;
```

Comment:
Is this necessary? Seems that passing `messages` to `useRef` is enough.

Reply:
the param passed to `useRef` only runs on component mount (not subsequent rerenders) so if I didn't have the second line, then the ref would never receive another update